src/ParametricUtils.jl

```diff
@@ -71,24 +74,61 @@ end
 
 Internal parametric extension to [`CalcFactor`](@ref) 
 """
-struct _CalcFactorParametric{CF}
+struct CalcFactorMahalanobis{CF}
   calcfactor!::CF
-  meanVal::Vector{Float64}
-  informationMat::Matrix{Float64}
+  varOrder::Vector{Symbol}
+  meas::Vector{Float64}
+  iΣ::Matrix{Float64}
 end
 
+function CalcFactorMahalanobis(fct::DFGFactor)
+  cf = getFactorType(fct)
+  varOrder = getVariableOrder(fct)
+  meas, iΣ = getParametricMeasurement(cf)
+  calcf = CalcFactor(cf, _getFMdThread(fct), 0, 0, (), [])
+  return CalcFactorMahalanobis(calcf, varOrder, meas, iΣ)
+end
 
 # pass in residual for consolidation with nonparametric
 # userdata is now at `cfp.cf.cachedata`
-function (cfp::_CalcFactorParametric)(variables...)
+function (cfp::CalcFactorMahalanobis)(variables...)
   # call the user function (be careful to call the new CalcFactor version only!!!)
-  res = zeros(length(cfp.meanVal))
-  cfp.calcfactor!(res, cfp.meanVal, variables...)
+  res = Vector{eltype(variables[1])}(undef, length(cfp.meas))
+  iΣ = cfp.iΣ
+  cfp.calcfactor!(res, cfp.meas, variables...)
```

Comment:
Yep, this is where the action is at (to consolidate nonparametric and parametric) and looks right to me, thanks!

PS, afterward we can worry about expanding both '(measurement..., variables...)'